### ISSUE TITLE
Rest#request can't cope with an empty response body

### ISSUE THREAD

**Issue by SimonWoolf** [ably/ably-js#676]
Not really a problem at the moment -- every endpoint that currently exists that we might use it for gives a response -- but still, a weird restriction and a spec violation (per HP1 it should be able to copy with an empty response)

**Comment by SimonWoolf** [comment]
> on what environment? NodeJS, browser, react native?

node

**Comment by owenpearson** [comment]
@vodolaz095 that link doesn't work for me but I know the test you're talking about. We've discussed it briefly [here](https://github.com/ably/ably-js/pull/804#discussion_r721583984) and it's just a bug in the backend API so we can ignore it. 

Unfortunately the tests are somewhat flakey so even when that's fixed the browser tests will still fail fairly often, you can run the tests locally with mocha HTML reporter by running `npm run test:webserver` and opening localhost:3000 in a browser of your choice. If you get a failing test there you can re-run it by clicking on the failing test so that's currently the best strategy for checking whether a test is consistently failing or just flakey.

**Comment by vodolaz095** [comment]
yes, i tried running tests in browser, but seems like i need to set credentials to access abbly servers in order to ran them properly, i'll research on it

[!https://scr.vodolaz095.life/s/LmUozzjbX.png](https://scr.vodolaz095.life/s/LmUozzjbX.png)
